idb/postgres/internal/ledger_for_evaluator/ledger_for_evaluator.go

```diff
@@ -124,258 +124,290 @@ func (l *LedgerForEvaluator) parseAccountTable(row pgx.Row) (basics.AccountData,
 		if err != nil {
-			return nil, fmt.Errorf("parseAssetTable() scan row err: %w", err)
+			return nil, fmt.Errorf("LookupWithoutRewards() err: %w", err)
 		}
 
-		if res == nil {
-			res = make(map[basics.AssetIndex]basics.AssetParams)
-		}
-		res[basics.AssetIndex(index)], err = encoding.DecodeAssetParams(params)
-		if err != nil {
-			return nil, fmt.Errorf("parseAssetTable() decode params err: %w", err)
+		if exists {
+			res[address] = lcAccountData
+		} else {
+			res[address] = nil
 		}
 	}
 
-	err := rows.Err()
+	err := results.Close()
 	if err != nil {
-		return nil, fmt.Errorf("parseAssetTable() scan end err: %w", err)
+		return nil, fmt.Errorf("LookupWithoutRewards() close results err: %w", err)
 	}
 
 	return res, nil
 }
 
-func (l *LedgerForEvaluator) parseAppTable(rows pgx.Rows) (map[basics.AppIndex]basics.AppParams, error) {
-	defer rows.Close()
-	var res map[basics.AppIndex]basics.AppParams
-
-	var index uint64
-	var params []byte
-
-	for rows.Next() {
-		err := rows.Scan(&index, &params)
-		if err != nil {
-			return nil, fmt.Errorf("parseAppTable() scan row err: %w", err)
-		}
+func (l *LedgerForEvaluator) parseAccountAssetTable(row pgx.Row) (basics.AssetHolding, bool /*exists*/, error) {
+	var amount uint64
+	var frozen bool
 
-		if res == nil {
-			res = make(map[basics.AppIndex]basics.AppParams)
-		}
-		res[basics.AppIndex(index)], err = encoding.DecodeAppParams(params)
-		if err != nil {
-			return nil, fmt.Errorf("parseAppTable() decode params err: %w", err)
-		}
+	err := row.Scan(&amount, &frozen)
+	if err == pgx.ErrNoRows {
+		return basics.AssetHolding{}, false, nil
 	}
-
-	err := rows.Err()
 	if err != nil {
-		return nil, fmt.Errorf("parseAppTable() scan end err: %w", err)
+		return basics.AssetHolding{}, false,
+			fmt.Errorf("parseAccountAssetTable() scan row err: %w", err)
 	}
 
-	return res, nil
+	assetHolding := basics.AssetHolding{
+		Amount: amount,
+		Frozen: frozen,
+	}
+	return assetHolding, true, nil
 }
 
-func (l *LedgerForEvaluator) parseAccountAppTable(rows pgx.Rows) (map[basics.AppIndex]basics.AppLocalState, error) {
-	defer rows.Close()
-	var res map[basics.AppIndex]basics.AppLocalState
-
-	var app uint64
-	var localstate []byte
-
-	for rows.Next() {
-		err := rows.Scan(&app, &localstate)
-		if err != nil {
-			return nil, fmt.Errorf("parseAccountAppTable() scan row err: %w", err)
-		}
+func (l *LedgerForEvaluator) parseAssetTable(row pgx.Row) (basics.Address /*creator*/, basics.AssetParams, bool /*exists*/, error) {
+	var creatorAddr []byte
+	var params []byte
 
-		if res == nil {
-			res = make(map[basics.AppIndex]basics.AppLocalState)
-		}
-		res[basics.AppIndex(app)], err = encoding.DecodeAppLocalState(localstate)
-		if err != nil {
-			return nil, fmt.Errorf("parseAccountAppTable() decode local state err: %w", err)
-		}
+	err := row.Scan(&creatorAddr, &params)
+	if err == pgx.ErrNoRows {
+		return basics.Address{}, basics.AssetParams{}, false, nil
```

Comment:
return err here?

Reply:
If the asset doesn't exist, there is no error. However, the third return value ("exists") is `false`.